src/site/content/en/blog/image-support-for-async-clipboard/index.md

```diff
@@ -0,0 +1,317 @@
+  const data = await fetch(imgURL);
+  const blob = await data.blob();
+  await navigator.clipboard.write([
+    new ClipboardItem(Object.defineProperty({}, blob.type, {
+      value: blob,
+      enumerable: true
+    }))
+  ]);
+  console.log('Image copied.');
+} catch(e) {
+  console.error(e, e.message);
+}
+```
+
+### Paste: reading an image from the clipboard {: #paste-image }
+
+The `navigator.clipboard.read()` method, which reads data from the clipboard, is
+also asynchronous, and Promise-based.
+
+To read an image from the clipboard, obtain a list of `ClipboardItem` objects,
+then iterate over them. Since everything is asynchronous, use the
+[`for ... of`][for-of] iterator, since it handles async/await code nicely.
+
+Each `ClipboardItem` can hold its contents in different types, so you'll
+need to iterate over the list of types, again using a `for ... of` loop.
+For each type, call the `getType()` method with the current type as an argument
+to obtain the corresponding image `Blob`. As before, this code is not tied
+to images, and will work with other future file types.
+
+```js
+async function getClipboardContents() {
+  try {
+    const clipboardItems = await navigator.clipboard.read();
+    for (const clipboardItem of clipboardItems) {
+      try {
+        for (const type of clipboardItem.types) {
+          const blob = await clipboardItem.getType(type);
+          console.log(URL.createObjectURL(blob));
+        }
+      } catch (e) {
+        console.error(e, e.message);
+      }
+    }
+  } catch (e) {
+    console.error(e, e.message);
+  }
+}
+```
+
+### Custom paste handler {: #custom-paste-handler }
+
+To dynamically handle paste events, listen for the `paste` event, call
+[`preventDefault()`][prevent-default] to prevent the default behavior
+in favor of your own logic, then use the code above to read the contents from
+the clipboard, and handle it in whatever way your app needs.
+
+```js
+document.addEventListener('paste', async (e) => {
+  e.preventDefault();
+  getClipboardContents();
+});
+```
+
+### Custom copy handler {: #custom-copy-handler }
+
+The [`copy` event][copy-event] includes a [`clipboardData`][clipboard-data]
+property with the items already in the right format, eliminating the need to
+manually create a blob. As before, don't forget to call `preventDefault()`.
+
+```js
+document.addEventListener('copy', async (e) => {
+  e.preventDefault();
+  try {
+    for (const item of e.clipboardData.items) {
+      await navigator.clipboard.write([
+        new ClipboardItem(Object.defineProperty({}, item.type, {
+          value: item,
+          enumerable: true
+        }))
+      ]);
+    }
+    console.log('Image copied.');
+  } catch(e) {
+    console.error(e, e.message);
+  }
+});
+```
+
+### Demo
+
+<iframe style="border: solid 1px;" width="100%" height="500"
+  src="https://jsfiddle.net/0794oysr/2/embedded/result,js,html,css"
+  allowfullscreen="allowfullscreen" frameborder="0">
+</iframe>
```

Comment:
@robdodson is it OK to embed a JSFiddle?

Side note: it's a very nice demo!

Reply:
I think we prefer Glitch if possible because you can edit it right on the site.

Reply:
Btw there are a lot of neat code snippets up above that could also be turned into interactive samples. Don't have to do it in this PR, but might be nice to think about as a future iteration.

Reply:
OK I converted it to Glitch and added the Glitch to our team project or whatever Glitch calls it: https://glitch.com/edit/#!/async-clipboard-api

Please replace the Fiddle with this Glitch, @petele:

```
<div class="glitch-embed-wrap">
  <iframe
    src="https://glitch.com/embed/#!/embed/async-clipboard-api?path=index.html&attributionHidden=true"
    alt="tabindex-zero on Glitch">
  </iframe>
</div>
```

Reply:
swapped over to kayce's glitch